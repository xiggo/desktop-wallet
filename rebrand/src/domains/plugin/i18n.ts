export const translations = {
	CATEGORIES: {
		GAME: "Game",
		GAMING: "Gaming",
		LANGUAGE: "Language",
		OTHER: "Other",
		UTILITY: "Utility",
	},

	DEVELOPER_RESPONSE: "Developer response",

	DISABLE_SUCCESS: `The plugin "{{name}}" has been disabled`,
	ENABLE_FAILURE: `Failed to enable plugin "{{name}}". Reason: {{msg}}.`,
	ENABLE_SUCCESS: `The plugin "{{name}}" has been enabled`,

	FILTERS: {
		CATEGORIES: {
			GAME: "Game",
			OTHER: "Other",
			UTILITY: "Utility",
		},
	},
	MANUAL_INSTALLATION_DISCLAIMER: {
		DISCLAIMER:
			"By accepting the following terms, you hereby assume the risk associated with downloading files and installing said files from a direct URL link. The aforementioned links have neither been tested nor approved by Solar.org.\n\nWe make no warranties, expressed or implied, as to the sustainability, availability, security, of these URLs. We are not responsible for any consequences which may occur from downloading, viewing, or installing files directly from a URL. Solar.org shall bear no liability for any loss suffered by users as a result of clicking an unverified direct URL.\n\nYou acknowledge that clicking unverified URLs may result in you downloading, viewing, or installing content that may have bugs, glitches, lack of functionality, or can cause damage to your machine or result in the loss of data.",
		TITLE: "Disclaimer",
	},
	MINIMUM_VERSION_NOT_SATISFIED:
		"In order to update this plugin please update the Solar Wallet to v{{minimumVersion}}+",

	MODAL_ENABLE_PLUGIN: {
		DESCRIPTION: "Would you like to enable the Plugin?",
		INSTALLATION: "Installation",
	},

	MODAL_INSTALL_PLUGIN: {
		ALLOW: "Allow",
		DESCRIPTION: "This plugin needs the following permissions:",
		DOWNLOAD_FAILURE: `Failed to download plugin "{{name}}"`,
		INSTALL_FAILURE: `Failed to install plugin "{{name}}". Reason: {{msg}}.`,
	},

	MODAL_MANUAL_INSTALL_PLUGIN: {
		DESCRIPTION: "Fetch the Plugin directly from GitHub by providing the repository URL below.",
		ERROR: "Failed to find a valid plugin repository. Please verify the URL and try again.",
		REPOSITORY_URL: "GitHub Repository URL",
		TITLE: "Install from URL",
	},

	MODAL_UNINSTALL: {
		DESCRIPTION:
			"Are you sure you want to remove '{{ name }}'? Uninstalling this plugin will remove it from all profiles on this system.",
		TITLE: "Uninstall Plugin",
	},

	MODAL_UPDATES_CONFIRMATION: {
		DESCRIPTION_COMPATIBLE: "The following plugins will be updated.",
		DESCRIPTION_INCOMPATIBLE:
			"Some of the plugins cannot be updated as they require a newer version of the Solar Wallet. Pressing Continue will install all compatible updates.",
		TITLE: "Plugin Updates",
	},

	NEW_UPDATES_AVAILABLE: "New updates available",

	NEW_VERSION_AVAILABLE: "New version available",

	PAGE_PLUGIN_MANAGER: {
		DESCRIPTION: "An easy way to find, install, and manage Plugins.",
		NO_PLUGINS_AVAILABLE: "This category has no available Plugins yet.",

		NO_PLUGINS_FOUND: `Your search query <strong>{{query}}</strong> does not match any Plugins in the manager.`,
		NO_PLUGINS_INSTALLED: "You haven't installed any Plugins yet. Once you install them, they will show up here.",
		TITLE: "Plugin Manager",

		VIEW: {
			ALL: "All",
			GAMING: "Gaming",
			LATEST: "Latest",
			MY_PLUGINS: "My Plugins",
			OTHER: "Other",
			SEARCH: "Search Results",
			UTILITY: "Utility",
		},
	},

	PERMISSIONS: {
		// Legacy permissions
		ALERTS: "Allows access to the Solar Wallet alerts",
		AUDIO: "Allows access to play audio from within the Solar Wallet",
		AVATARS: "Plugin contains custom avatars",
		COMPONENTS: "Allows loading custom components",
		DIALOGS: "Allows using file dialogs",
		EVENTS: "Allows access to the Solar Wallet events",
		FILESYSTEM: "Allows using file dialogs",
		HTTP: "Allows performing external web requests",
		LANGUAGES: "Allows loading additional languages for the Solar Wallet",
		LAUNCH: "Allows to register a custom view",
		MENU_ITEMS: "Allows adding custom menu items to the Solar Wallet sidebar",
		MESSAGING: "Allows WebFrames access to a one-way messaging system",
		PEER_ALL: "Allows access to the peer discovery",
		PEER_CURRENT: "Allows access to the currently connected peer",
		PROFILE: "Allows access to the currently active profile",
		PROFILE_ALL: "Allows access to all available profiles",
		PROFILE_CURRENT: "Allows access to the currently active profile",
		PUBLIC: "Allows navigation to wallet routes and provides access to the Font Awesome icon set",
		ROUTES: "Allows loading additional routes into the Solar Wallet",
		STORAGE: "Allows storing data within the Solar Wallet, using key-value pairs",
		STORE: "Allows storing data within the Solar Wallet, using key-value pairs",
		THEME: "Allows loading additional custom themes for the Solar Wallet",
		THEMES: "Allows loading additional custom themes for the Solar Wallet",
		TIMERS: "Allows using timeouts and intervals",
		UI_COMPONENTS: "Allows access to the standard Solar Wallet components used throughout",
		UTILS: "Allows using utilities such as the BigNumber type and dayjs",
		WALLET_TABS: "Allows showing an additional tab/page on the Wallet screen",
		WEBFRAME: "Allows showing remote URL pages within a frame",
		WEBSOCKET: "Allows connections to websockets",
	},

	PLUGIN_INFO: {
		ABOUT: "About",
		DISCLAIMER:
			"The availability of this Plugin in the Solar Wallet does not mean that Solar.org is directly involved in its development or developers. By installing it, you assume any associated risks.",
		NOT_ENABLED: "This Plugin is not enabled",
		NOT_LAUNCHEABLE: "This Plugin cannot be launched",
		REPORT: "Report Plugin",
		REQUIREMENTS: "Requirements",
		SCREENSHOTS: "Screenshots",
		WALLET_VERSION: "Solar Wallet Version v{{minimumVersion}}+",
	},

	REQUIRED_VERSION: "Required Version",

	STATUS: {
		COMPATIBLE: "Compatible",
		DISABLED: "Disabled",
		ENABLED: "Enabled",
		INCOMPATIBLE: "Incompatible",
		NOT_INSTALLED: "Not installed",
	},

	SUCCESSFULLY_INSTALLED: "Successfully Installed",

	UPDATE_ALL: "Update All",

	UPDATE_ALL_NOTICE_INCOMPATIBLE_one:
		"There is {{count}} update available for your plugins, but your wallet version is incompatible. Please update the Solar Wallet to continue.",
	UPDATE_ALL_NOTICE_INCOMPATIBLE_other:
		"There are {{count}} updates available for your plugins, but your wallet version is incompatible. Please update the Solar Wallet to continue.",
	UPDATE_ALL_NOTICE_one: "There is {{count}} update available for your plugins. Would you like to update it?",
	UPDATE_ALL_NOTICE_other:
		"There are {{count}} updates available for your plugins. Would you like to update them all now?",

	WARNING_DISCLAIMER:
		"Please make sure to check the documentation of the Plugin before installing it. By installing the Plugin, you assume any associated risks.",
};
