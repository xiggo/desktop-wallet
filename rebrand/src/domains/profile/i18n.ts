export const translations = {
	CREATE_PROFILE: "Create Profile",
	IMPORT: {
		ERROR_STEP: {
			DESCRIPTION:
				"An error occurred while importing the profile, try importing again. Select Retry to re-import the profile. If the error occurs again, you can contact our support team.",
		},
		FORM_STEP: {
			DESCRIPTION: "Your profile addresses have been imported.",
		},

		PASSWORD_DESCRIPTION:
			"The profile you are importing contains a password that has been set for security purposes. Enter the password to confirm.",
		PASSWORD_TITLE: "Profile Password",
		PROCESSING_IMPORT_STEP: {
			DESCRIPTION: "The file with name {{name}} is being imported.",
			PLEASE_WAIT: "Please wait...",
		},
		SELECT_FILE_STEP: {
			BROWSE_FILES: "Browse Files",
			CLICK_HERE: "Click here.",
			DEPRECATION_WARNING:
				"Please note that importing a Profile from Desktop Wallet v2 in .json format is supported temporarily and will soon be disabled. We recommend that you re-save your Profile in the new .dwe format.",
			DESCRIPTION:
				"Select a {{fileFormat}} file with your Profile and related properties to start the import process.",
			DRAG_AND_DROP: "Drag & Drop or",
			ERRORS: {
				NOT_SUPPORTED:
					"This file type is not supported.\nPlease retry with a <strong>{{fileFormat}}</strong> file.",
				TOO_MANY: "You tried to import <strong>{{fileCount}}</strong> files.\nPlease retry with one file only.",
			},
			LEGACY_IMPORT: "Importing from Desktop Wallet v2?",
			SUPPORTED_FORMAT: "Supported format is {{fileFormat}}",
			UPLOAD_TITLE: "Click here to upload",
		},
		TITLE: "Import Profile",
	},
	MODAL_DELETE_PROFILE: {
		DESCRIPTION: "Do you really want to delete this profile? Once deleted, you will not be able to restore it.",
		TITLE: "Delete Profile",
	},
	MODAL_HISTORY: {
		TITLE: "History",
		TYPES: {
			REGISTRATION: "Registration",
			RESIGN: "Resign",
			UPDATE: "Update",
		},
	},
	MODAL_RESET_PROFILE: {
		DESCRIPTION:
			"By performing this action, all of your Profile settings will be restored to default. This won't delete your wallets or plugins. This action cannot be undone. Do you want to reset your settings?",
		SUCCESS: "Your profile has been reset successfully",
		TITLE: "Reset Profile Settings",
	},
	MODAL_SELECT_SENDER: {
		DESCRIPTION: "Find and select the sender from your wallets",
		SEARCH_PLACEHOLDER: "Enter name or address",
		TITLE: "Select Sender",
	},
	MODAL_SIGN_IN: {
		DESCRIPTION: "Sign in to access your Profile.",
		MAX_ATTEMPTS_ERROR:
			"Maximum sign in attempts reached - please wait {{remainingTime}} seconds before trying again.",
		TITLE: "Sign In",
	},
	MODAL_WELCOME: {
		DONT_SHOW_CHECKBOX_LABEL: "Don't show this quick tour again.",
		STEP_1_DESCRIPTION:
			"Start a quick tour to learn how to get the most out of the Solar Wallet, or skip it to go directly to your profile.",
		STEP_1_TITLE: "Welcome to Solar Wallet",
		STEP_2_DESCRIPTION:
			"Create or import existing wallets from a broad range of assets and secure them using industry-standard encryption.",
		STEP_2_TITLE: "Manage Your Assets",
		STEP_3_DESCRIPTION:
			"Solar lets you extend your wallet’s functionality with a growing collection of Games, Utilities and more.",
		STEP_3_TITLE: "Enhance Your Experience With Plugins",
		STEP_4_DESCRIPTION: "With Solar, you’re in control. Swap assets effortlessly without ever leaving your wallet.",
		STEP_4_TITLE: "Exchange Assets On The Fly",
		STEP_5_DESCRIPTION:
			"Customize or search your news feed to view the latest updates on your favorite cryptoassets.",
		STEP_5_TITLE: "Stay Up To Date",
	},
	PAGE_CREATE_PROFILE: {
		DESCRIPTION: "Create a new Profile below.",
		TITLE: "Create Profile",
		VALIDATION: {
			NAME_EXISTS: "A profile with this name already exists",
		},
	},
	PAGE_WELCOME: {
		HAS_EXPORTED_PROFILES: "Exported a Profile before?",
		IMPORT_PROFILE: "Import it here.",
		IMPORT_PROFILE_TITLE: "Import Profile",
		WITHOUT_PROFILES: {
			DESCRIPTION: "Create a new Profile to get started.",
			TITLE: "Create Profile",
		},
		WITH_PROFILES: {
			DESCRIPTION: "Choose from an existing Profile below or create a new Profile to get started.",
			TITLE: "Select Profile",
		},
	},
};
