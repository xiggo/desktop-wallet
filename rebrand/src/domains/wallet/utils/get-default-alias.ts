import { Networks } from "@payvo/sdk";
import { Contracts } from "@payvo/sdk-profiles";

interface GetDefaultAliasInput {
	profile: Contracts.IProfile;
	network: Networks.Network;
}

export const getDefaultAlias = ({ profile, network }: GetDefaultAliasInput): string => {
	const makeAlias = (count: number) => `${network.displayName()} #${count}`;

	const sameCoinWallets = profile.wallets().findByCoinWithNetwork(network.coin(), network.id());

	let counter = sameCoinWallets.length;

	if (counter === 0) {
		counter = 1;
	}

	while (profile.wallets().findByAlias(makeAlias(counter))) {
		counter++;
	}

	return makeAlias(counter);
};
