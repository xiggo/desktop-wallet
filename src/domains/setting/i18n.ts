export const translations = {
	APPEARANCE: {
		MENU_ITEM: "Appearance",
		OPTIONS: {
			ACCENT_COLOR: {
				COLORS: {
					BLUE: "Blue",
					GREEN: "Green",
				},
				DESCRIPTION: "Choose a color for buttons and other wallet features.",
				TITLE: "Accent Color",
			},
			EXPANDED_TABLES: {
				DESCRIPTION: "Increases spacing of table items.",
				TITLE: "Expanded Tables",
			},
			LATEST_TRANSACTIONS: {
				DESCRIPTION: "Enables latest transactions on Portfolio page.",
				TITLE: "Portfolio Latest Transactions",
			},
			VIEWING_MODE: {
				DESCRIPTION: "Choose a light or dark viewing mode.",
				TITLE: "Viewing Mode",
				VIEWING_MODES: {
					DARK: "Dark",
					LIGHT: "Light",
				},
			},
			WALLET_NAMING: {
				DESCRIPTION: "Prioritize showing network provided names over wallet names.",
				TITLE: "Wallet Naming",
			},
		},
		SUBTITLE: "Customize the look of the application.",
		TITLE: "Appearance",
	},

	EXPORT: {
		ADD_INFORMATION_ABOUT_THE_NETWORK: "Add information about the network",
		MENU_ITEM: "Export",
		OPTIONS: {
			EXCLUDE_EMPTY_WALLETS: {
				DESCRIPTION: "Wallets with a balance of 0 will not be exported.",
				TITLE: "Exclude Empty Wallets",
			},
			EXCLUDE_LEDGER_WALLETS: {
				DESCRIPTION: "Ledger hardware addresses will not be exported.",
				TITLE: "Exclude Ledger Wallets",
			},
		},
		SAVE_GENERAL_CUSTOMIZATIONS: "Save general settings customizations",
		SUBTITLE:
			"Your exported Profile will not contain your mnemonic passphrases, only addresses and respective names.",
		SUCCESS: "Your settings have been exported successfully",
		TITLE: "Export Profile",
	},

	GENERAL: {
		ERROR: "Your profile settings could not be updated",
		MENU_ITEM: "General",
		OTHER: {
			DARK_THEME: {
				DESCRIPTION: "Enables a dark, high contrast scheme.",
				TITLE: "Dark Theme",
			},
			DEVELOPMENT_NETWORKS: {
				DESCRIPTION:
					"To create or import wallets from development and test networks, you must first enable this feature.",
				TITLE: "Development and Test Networks",
			},
			ERROR_REPORTING: {
				DESCRIPTION: "Automatically send reports to help to improve our products.",
				TITLE: "Error Reporting",
			},
			TITLE: "Other",
		},
		PERSONAL: {
			CONFIRM_PASSWORD: "Confirm Password",
			CURRENCY: "Currency",
			LANGUAGE: "Language",
			MARKET_PROVIDER: "Price Oracle",
			NAME: "Profile Name",
			PASSPHRASE_LANGUAGE: "Passphrase Language",
			PASSWORD: "Password",
			PROFILE_IMAGE: "Select Profile Image",
			REMOVE_AVATAR: "Remove Avatar",
			TIME_FORMAT: "Time Format",
			TITLE: "Personal Details",
			UPLOAD_AVATAR: "Upload Avatar",
			VALIDATION: {
				NAME_EXISTS: "Profile name already exists",
			},
		},
		SECURITY: {
			ADVANCED_MODE: {
				DESCRIPTION: "Allows installations from direct URLs. Enable at your own risk.",
				TITLE: "Advanced Mode",
			},
			AUTOMATIC_SIGN_OUT_PERIOD: {
				TITLE: "Auto Log-Off",
			},
			SCREENSHOT_PROTECTION: {
				DESCRIPTION: "When enabled, wallet application is hidden from screen-capturing software.",
				TITLE: "Screenshot Protection",
			},
			TITLE: "Security",
		},
		SUBTITLE: "Customize your wallet to suit your needs.",
		SUCCESS: "Your profile settings have been updated",
		TITLE: "Profile Settings",
		UNSUPPORTED_CURRENCY: "{{currency}} is not supported by {{provider}}. Currency has been changed to USD.",
	},

	MODAL_DEVELOPMENT_NETWORK: {
		DESCRIPTION:
			"Disabling this setting will hide your wallets associated with development networks. Are you sure you want to continue?",
		TITLE: "Development Network",
	},

	PASSWORD: {
		BUTTON: {
			CREATE: "Set Password",
			REMOVE: "Remove Password",
			UPDATE: "Change Password",
		},
		CURRENT: "Current Password",
		ERROR: {
			FALLBACK: "Something went wrong.",
			MISMATCH: "The current password does not match.",
		},
		MENU_ITEM: "Password",
		PASSWORD_1: "New Password",
		PASSWORD_2: "Confirm New Password",
		REMOVAL: {
			DESCRIPTION:
				"Removing your profile password will result in anyone with access to your Solar Wallet to see all your profile data. Are you sure you wish to remove it?",
			PROFILE_PASSWORD: "Profile Password",
			SUCCESS: "Password successfully removed.",
			TITLE: "Remove Profile Password",
		},
		SUBTITLE: {
			CREATE: "Set your password below to secure your Profile.",
			UPDATE: "Change the password used to secure your Profile below.",
		},
		SUCCESS: "Your password has been successfully changed.",
		TITLE: "Profile Password",
	},

	PLUGINS: {
		ADD_PLUGIN: "Add Plugin",
		MENU_ITEM: "Plugins",
		PLUGIN_SOURCE: {
			DESCRIPTION: "Allows Plugins to be installed from third-party sources.",
			LOAD_FROM: "Load from",
			TITLE: "Plugin Source",
		},
		SUBTITLE: "Manage your Plugin settings below.",
		SUCCESS: "Your plugin settings have been updated",
		TITLE: "Plugin Settings",
	},

	TITLE: "Settings",
};
