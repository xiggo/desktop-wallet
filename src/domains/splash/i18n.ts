export const translations = {
	BRAND: "Solar Wallet",
	COPYRIGHT: "Â© solar.network",
	LOADING: "Initializing...",
	PRODUCT: "A Payvo Product",
	RIGHTS: "All Rights Reserved",
	VERSION: "Version",
};
