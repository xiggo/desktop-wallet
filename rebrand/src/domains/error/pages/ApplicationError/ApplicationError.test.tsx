import userEvent from "@testing-library/user-event";
import React from "react";

import { ApplicationError } from "./ApplicationError";
import { translations } from "@/domains/error/i18n";
import { render, screen } from "@/utils/testing-library";

describe("ApplicationError", () => {
	it("should render", () => {
		const onResetErrorBoundary = jest.fn();
		const { asFragment, container } = render(<ApplicationError resetErrorBoundary={onResetErrorBoundary} />);

		expect(container).toBeInTheDocument();
		expect(screen.getByTestId("ApplicationError__text")).toHaveTextContent(translations.APPLICATION.TITLE);
		expect(screen.getByTestId("ApplicationError__text")).toHaveTextContent(translations.APPLICATION.DESCRIPTION);

		userEvent.click(screen.getByTestId("ApplicationError__button--reload"));

		expect(onResetErrorBoundary).toHaveBeenCalledWith(
			expect.objectContaining({ nativeEvent: expect.any(MouseEvent) }),
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
