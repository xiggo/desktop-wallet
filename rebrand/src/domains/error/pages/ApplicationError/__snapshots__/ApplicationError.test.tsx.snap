// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationError should render 1`] = `
<DocumentFragment>
  <main
    class="theme-dark"
    data-testid="Main"
  >
    <div
      class="flex relative flex-col min-h-screen"
    >
      <nav
        aria-labelledby="main menu"
        class="NavigationBarblocks__NavWrapper-sc-nciu2a-0 cxmxRW"
      >
        <div
          class="flex relative h-21"
        >
          <div
            class="flex flex-1 px-8 ml-12"
          >
            <div
              class="flex items-center my-auto"
            >
              <div
                class="flex items-center justify-center my-auto mr-4 pl-0.5 text-white rounded-xl w-11 h-11 bg-theme-success-600"
              >
                <svg
                  height="28"
                >
                  payvo-logo.svg
                </svg>
              </div>
            </div>
          </div>
        </div>
      </nav>
      <div
        class="flex flex-col flex-1"
      >
        <div
          class="Section__SectionWrapper-sc-plnv9l-0 fYoPaw flex flex-col flex-1 justify-center text-center"
        >
          <div
            class="container px-10 mx-auto"
          >
            <div
              class="flex justify-center mx-auto max-w-lg"
            >
              <svg>
                generic-error-dark-green.svg
              </svg>
            </div>
            <div
              class="mt-8"
              data-testid="ApplicationError__text"
            >
              <h2
                class="text-2xl font-bold"
              >
                Something went wrong
              </h2>
              <p
                class="text-theme-secondary-text"
              >
                It looks like something went wrong. Try visiting another page or reloading.
              </p>
            </div>
            <button
              class="OriginalButton__StyledButton-sc-o0fs2j-0 isLyLB mt-8"
              data-testid="ApplicationError__button--reload"
              type="button"
            >
              <div
                class="flex items-center space-x-2"
              >
                Reload
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
